### PR TITLE
wdc 1.0.0 (new formula)

### PR DESCRIPTION
**Pull request by designerror**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

---

### CODE REVIEW

**Review by mistydemeo — DISMISSED** [review]

File: Formula/wdc.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  depends_on "curl"
+  depends_on "pugixml"
+
+  def install
+    system "cmake", ".", *std_cmake_args
+    system "make", "install"
+  end
+
+  test do
+    system "true"
```
Comment (designerror) [inline Formula/wdc.rb]: I added test.

**Comment by designerror** [comment]
Fixed test for `linuxbrew` on Ubuntu.

File: Formula/wdc.rb [hunk 2]
```diff
@@ -0,0 +1,39 @@
+class Wdc < Formula
+  desc "WebDAV Client provides easy and convenient to work with WebDAV-servers."
+  homepage "https://designerror.github.io/webdav-client-cpp"
+  url "https://github.com/designerror/webdav-client-cpp/archive/v1.0.0.tar.gz"
+  sha256 "649a75a7fe3219dff014bf8d98f593f18d3c17b638753aa78741ee493519413d"
+
+  depends_on "cmake" => :build
+  depends_on "curl"
```
Comment (MikeMcQuaid) [inline Formula/wdc.rb]: Can this not use the system version?

File: Formula/wdc.rb [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+          {"webdav_hostname", "https://webdav.example.com"},
+          {"webdav_login",    "webdav_login"},
+          {"webdav_password", "webdav_password"}
+        };
+        std::shared_ptr<WebDAV::Client> client(WebDAV::Client::Init(options));
+        auto check_connection = client->check();
+        assert(!check_connection);
+      }
+    EOS
+    system ENV.cc, "test.cpp", "-L#{lib}", "-lwebdavclient", "-lpthread", "-lpugixml", "-lm", "-lcurl", "-lssl", "-lcrypto", "-lstdc++", "-std=c++11", "-o", "test"
```
Comment (MikeMcQuaid) [inline Formula/wdc.rb]: Please split this onto multiple lines at 80 characters.

**Review by MikeMcQuaid — CHANGES_REQUESTED** [review]

File: Formula/wdc.rb (hunk 2 above)
Comment (designerror) [inline Formula/wdc.rb]: necessary for `linuxbrew`.

File: Formula/wdc.rb (hunk 3 above)
Comment (designerror) [inline Formula/wdc.rb]: I split.

**Comment by designerror** [comment]
`brew audit https://raw.githubusercontent.com/designerror/homebrew-core/master/Formula/wdc.rb --new-formula` - ok
`brew audit https://raw.githubusercontent.com/designerror/homebrew-core/master/Formula/wdc.rb --new-formula` - ok

File: Formula/wdc.rb (hunk 2 above)
Comment (MikeMcQuaid) [inline Formula/wdc.rb]: Homebrew/homebrew-core doesn't support Linuxbrew so please remove it, thanks.

Comment (designerror) [inline Formula/wdc.rb]: you are talking about `cmake` or `curl` or both ?

**Comment by designerror** [comment]
I tried to fix test for `linuxbrew` on `CentOS`. But it is not work for `homebrew`. I made it back.

**Comment by designerror** [comment]
I finished the refactoring of the formula.

**Review by MikeMcQuaid — APPROVED** [review]

**Comment by designerror** [comment]
@mistydemeo, I finished the refactoring of the formula.

**Comment by designerror** [comment]
@mistydemeo, I finished the refactoring of the formula.

**Comment by designerror** [comment]
@mistydemeo, I added `pc` file.

**Comment by MikeMcQuaid** [comment]
Thanks for your contribution to Homebrew! Without people like you submitting PRs we couldn't run this project. You rock!

File: Formula/wdc.rb [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+    exec_prefix=${prefix}
+    libdir=${exec_prefix}/lib
+    includedir=${exec_prefix}/include
+    Name: libwebdavclient
+    Description: Modern and convenient C++ WebDAV Client library
+    Version: 1.0.0
+    Libs: -L${libdir} -lwebdavclient
+    Libs.private: -lpthread -lpugixml -lm -lcurl -lssl -lcrypto
+    Cflags: -I${includedir}
+    EOS
```
Comment (MikeMcQuaid) [inline Formula/wdc.rb]: @designerror Didn't want to block this PR being merged but a `.pc` file should really be generated and installed by your CMake buildsystem. Could you do that for a future version? Thanks!

**Pull request closed without merging**